core/src/main/java/org/keycloak/crypto/JavaAlgorithm.java

```diff
@@ -96,6 +110,18 @@ public static String getJavaAlgorithmForHash(String algorithm) {
                 return SHA384;
             case Algorithm.PS512:
                 return SHA512;
+            case Algorithm.EdDSA:
+                if (curve != null) {
+                    switch (curve) {
+                        case Algorithm.Ed25519:
+                            return SHA256;
+                        case Algorithm.Ed448:
+                            return SHA384;
+                        default:
+                            throw new IllegalArgumentException("Unknown curve for EdDSA " + curve);
+                    }
+                }
+                return SHA256;
```

Comment:
It seems that Ed25519 uses hash SHA-512 and Ed448 uses SHAKE-256 (see [here](https://bitbucket.org/openid/connect/issues/1125)). Can you please change that accordingly?

```diff
diff --git a/core/src/main/java/org/keycloak/crypto/JavaAlgorithm.java b/core/src/main/java/org/keycloak/crypto/JavaAlgorithm.java
index ca38778518..3ee487e437 100644
--- a/core/src/main/java/org/keycloak/crypto/JavaAlgorithm.java
+++ b/core/src/main/java/org/keycloak/crypto/JavaAlgorithm.java
@@ -37,6 +37,7 @@ public class JavaAlgorithm {
     public static final String SHA256 = "SHA-256";
     public static final String SHA384 = "SHA-384";
     public static final String SHA512 = "SHA-512";
+    public static final String SHAKE256 = "SHAKE-256";
 
     public static String getJavaAlgorithm(String algorithm) {
         return getJavaAlgorithm(algorithm, null);
@@ -114,14 +115,14 @@ public class JavaAlgorithm {
                 if (curve != null) {
                     switch (curve) {
                         case Algorithm.Ed25519:
-                            return SHA256;
+                            return SHA512;
                         case Algorithm.Ed448:
-                            return SHA384;
+                            return SHAKE256;
                         default:
                             throw new IllegalArgumentException("Unknown curve for EdDSA " + curve);
                     }
                 }
-                return SHA256;
+                return SHA512;
             case Algorithm.AES:
                 return AES;
             default:
```

Reply:
Thank you for your suggestion. I read https://bitbucket.org/openid/connect/issues/1125 and confirmed that Ed25519 uses SHA512 (SHA-2) while Ed448 uses SHAKE-256 (SHA-3). I fixed the codes as you suggested.